### ISSUE TITLE
[🐞] Components no longer working as expected in Qwik 0.101.0 and QwikCity 0.101.0

### ISSUE THREAD

**Issue by jsaad-syncoms** [BuilderIO/qwik#3749]
### Which component is affected?

Qwik Runtime

### Describe the bug

I cannot even write a basic component without getting the following error - TS2345:

Argument of type '() => JSX.Element' is not assignable to parameter of type 'OnRenderFn<{}>'.
  Call signature return types 'Element' and 'JSXNode<any> | null' are incompatible.
    The types of 'dev' are incompatible between these types.
      Type 'DevJSX | undefined' is not assignable to type 'DevJSX | undefined'. Two different types with this name exist, but they are unrelated.
        Type 'DevJSX' is missing the following properties from type 'DevJSX': ctx, isStaticts(2345)

### Reproduction

Cannot even build project so cannot provide reproduction

### Steps to reproduce

try building a component

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 476.42 MB / 7.80 GB
  Binaries:
    Node: 18.14.2 - C:\Program Files\nodejs\node.EXE
    npm: 9.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 111.0.5563.147
    Edge: Spartan (44.22621.1413.0), Chromium (112.0.1722.39)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @builder.io/partytown: ^0.7.6 => 0.7.6
    @builder.io/qwik: 0.101.0 => 0.101.0
    @builder.io/qwik-city: 0.101.0 => 0.101.0
    undici: 5.21.2 => 5.21.2
    vite: 4.2.1 => 4.2.1
```

### Additional Information

Not sure why this happening - even copy-pasting example components directly from the Qwik Docs causes this error

Example copied from https://qwik.builder.io/docs/components/overview/#inline-components:
<img width="244" alt="image" src="https://user-images.githubusercontent.com/46931121/231448965-954e7e10-1df1-4068-8d2a-a90e754f26b1.png">
![image](https://user-images.githubusercontent.com/46931121/231449204-742654ad-2345-4d91-8b17-46f46ac1666d.png)

This method doesn't seem to produce an error but I don't know why I would need to do this instead of doing what I've always been doing:
<img width="242" alt="image" src="https://user-images.githubusercontent.com/46931121/231449663-7f8e6bcc-b0bf-4e20-a888-079d308f3001.png">

**Comment by jsaad-syncoms** [comment]
**EDIT:** I've reinstalled Qwik 0.21.0 and Qwik City 0.5.3 to be able to resume work on my project but I would obviously like to be able to update and use the latest version. If someone can help me understand what's happening, that would be a great help.

**Comment by jsaad-syncoms** [comment]
**EDIT:** For anyone else experiencing this, the issue is qwik-city, not qwik. I changed qwik-city in package.json back to 0.5.3, did an npm install and that fixed everything. I then changed qwik-city "~0.101.0" and did another npm install and thatgot me to the latest version without any errors.

I had done the original update using the "ncu" command (using the npm-check-updates package), this updated qwik-city to "0.101.0" instead of "~0.101.0", hence the issues described above.
